src/api/controllers/likeDislikeController.js

```diff
@@ -0,0 +1,88 @@
+import models from '../models';
+import statusCode from '../../helpers/constants/status.codes';
+/**
+ * this is the LikeDislikeController controller
+ *
+ * @export
+ * @class LikeDislikeControllerController
+ */
+
+/**
+ * containing all like dislike comment 's model controllers
+ * @export
+ * @class LikeDislikeController
+ */
+export default class LikeDislikeController {
+  /**
+   * register a new
+   * @static
+   * @param {*} req the request
+   * @param {*} res the response to be sent
+   * @memberof LikeDislikeController
+   * @returns {Object} res
+   */
+  static async likeComment(req, res) {
+    const { commentId } = req.params;
+    const { id } = req.user.user;
+    const hasDisliked = await models.LikeDislike.findAll({
+      where: {
+        commentId,
+        userId: id,
+        dislikes: 1,
+        likes: 0
+      }
+    });
+    if (hasDisliked[0]) {
+      await models.LikeDislike.update({ dislikes: 0, likes: 1 },
+        { where: { id: hasDisliked[0].id } });
+      return res.status(statusCode.OK).json({
+        message: 'Thank you for reacting on  this comment!'
+      });
+    }
+    await models.LikeDislike.create({
+      userId: id,
+      commentId,
+      dislikes: 0,
+      likes: 1
+    });
+
+    return res.status(statusCode.CREATED).json({
+      message: 'Thank you for reacting on  this comment!'
+    });
+  }
+
+  /**
+   * @description - Users should be able to dislike a comment
+   * @param {Object} req - Request Object
+   * @param {Object} res  - Response Object
+   * @returns {Object} - Response object
+   */
+  static async dislikeComment(req, res) {
+    const { commentId } = req.params;
+    const { id } = req.user.user;
+    const hasLiked = await models.LikeDislike.findAll({
+      where: {
+        commentId,
+        userId: id,
+        likes: 1
+      }
+    });
+    if (hasLiked[0]) {
```

Comment:
same as here. I think you can user middlewares when checking if the user liked the comment, 
maybe, you can explain more about this.